src/key.rs

```diff
@@ -213,7 +213,32 @@ impl SecretKey {
     }
 }
 
-serde_impl!(SecretKey, constants::SECRET_KEY_SIZE);
+#[cfg(feature = "serde")]
+impl ::serde::Serialize for SecretKey {
+    fn serialize<S: ::serde::Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        if s.is_human_readable() {
+            s.collect_str(self)
+        } else {
+            s.serialize_bytes(&self[..])
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> ::serde::Deserialize<'de> for SecretKey {
+    fn deserialize<D: ::serde::Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+        if d.is_human_readable() {
+            d.deserialize_str(super::serde_util::FromStrVisitor::new(
```

Comment:
Perhaps remove `super::` and add a use statement that imports using the fully qualified path (although I would remove `serde_util::` also). Then we can remove `super::` at no additional cost all the way through this PR. (I'm still singing the same old tune :)

Reply:
I am not sure how useful a `use` statement will be.

This code is feature-gated so I would either have to feature-gate the `use` statement at the top of the file as well or accept that rustc will issue a dead-code warning if we compile without the `serde` feature flag.

For feature-gated code, I usually err on the side of having full-qualified paths, esp. if they only occur once within the code that is gated.

We could potentially include a `use super::serde_util;` at the beginning of the function and then reference both `FromStrVisitor` and `BytesVisitor` as `serde_util::FromStrVisitor`.

Reply:
Thanks for the response, I didn't catch the feature gated nuance when I commented.  Not sure which part of the comment @sgeisler is ok'ing but I like use statement inside the function option. Its pretty minor though, so don't sweat it :)